x-pack/platform/plugins/shared/embeddable_alerts_table/public/embeddable_alerts_table_factory.test.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+// Write a test that verifies that the `AlertsTableEmbeddable` component renders the `AlertsTable` component with the correct props.
+
+import React from 'react';
+import { render } from '@testing-library/react';
+import type { StartDependencies } from './types';
+import { coreMock } from '@kbn/core/public/mocks';
+import { getEmbeddableAlertsTableFactory } from './embeddable_alerts_table_factory';
+import { LOCAL_STORAGE_KEY_PREFIX } from './constants';
+
+const core = coreMock.createStart();
+jest.mock('@kbn/response-ops-alerts-table', () => ({
+  AlertsTable: jest.fn(() => <div data-test-subj="alertsTable" />),
+}));
+const { AlertsTable: mockAlertsTable } = jest.requireMock('@kbn/response-ops-alerts-table');
+
+const uuid = 'test-uuid';
+
+describe('getEmbeddableAlertsTableFactory', () => {
+  const factory = getEmbeddableAlertsTableFactory(core, {} as StartDependencies);
+
+  it('renders AlertsTable with correct props', async () => {
+    const { Component } = await factory.buildEmbeddable(
+      {
+        timeRange: {
+          from: '2025-01-01T00:00:00.000Z',
+          to: '2025-01-01T01:00:00.000Z',
+        },
+        title: 'Test embeddable alerts table',
+      },
+      (apiRegistration) => apiRegistration as any,
+      uuid,
+      // These are unused by our factory
+      {} as any,
+      () => ({} as any),
+      {} as any
+    );
+
+    render(<Component />);
+
+    expect(mockAlertsTable).toHaveBeenCalledWith(
```

Comment:
Let's add one, except where we check that the table was rendered correctly like `expect(screen.getByTestId('alertsTable')).toBeInTheDocument();`